### ISSUE TITLE
Automate water rights data retrieval from states’ datasets to WaDE

### ISSUE THREAD

**Issue by amabdallah** [WSWCWaterDataExchange/MappingStatesDataToWaDE2.0#255]
One of the most frequently asked questions was: how often do we update the data? People dont want to work with outdated data.

Let's revisit this and try to set it up for a single state as a test.
In layman terms, we need to host the Python code in Azure and let it export the CSV file into the blog storage, then trigger the data uploader. We once set a simple pipeline before (probably in 2019 or 2020) where, in the end, it sends an email saying it successfully imported the data. Let's look into it and we could ask DPL for their advise.
https://learn.microsoft.com/en-us/azure/architecture/data-guide/relational-data/etl

### Adel wrote this in Sept 2021.
To import water rights datasets to WaDE, we have thus far manually downloaded those water rights datasets to a local machine, followed by running a custom Python script to map and fit the state's data to WaDE generalized data template, then manually running the WaDE data loader script to upload the data into WaDE database in Microsoft Azure (Figure 4). In this sub-task, we aim at automating these processes with an automatic download, process, and upload of data to the WaDE database regularly. This will help reduce the effort it takes to share the data with WaDE and to keep WaDE up to date with any changes published to the states’ datasets. This automation is possible through the Azure Workflow Automation service.
We are continuously coordaining with the point of contact for our member states to review the mapping of their data into WaDE. We plan to have them review our documentation and ask if they approve of how their data appears in the WaDE API and the WestDAAT dashboard (this is done before we officially release a state’s data). 

![image](https://github.com/WSWCWaterDataExchange/MappingStatesDataToWaDE2.0/assets/3268971/cb8378e7-1daf-4009-abc6-c240d1511ece)
Four steps to be automated for extracting, transforming, and loading data from our states to the WaDE database

**Comment by rwjam** [comment]
can try this: https://github.com/Nicceboy/python-markdown-generator
